packages/stepper/src/StepLabel.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { Body } from '@leafygreen-ui/typography';
+import { PropsWithChildren } from 'react';
+import { StepLabelProps, StepStates } from './types';
+import { palette } from '@leafygreen-ui/palette';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { useStepperContext } from './StepperContext';
+import { stepLabelClassName } from './constants';
+
+const StepLabel = ({ children, state }: PropsWithChildren<StepLabelProps>) => {
+  const isCurrent = state === StepStates.Current;
+  const { isDarkMode } = useStepperContext();
+
+  const completedMultipleStyles = css`
+    color: ${isDarkMode ? palette.green.base : palette.green.dark2};
+    text-decoration-line: underline;
+    text-decoration-style: dotted;
+    text-underline-position: under;
+  `;
+
+  const completedStyles = css`
+    color: ${isDarkMode ? palette.green.base : palette.green.dark2};
+  `;
+
+  const currentStyles = css`
+    color: ${isDarkMode ? palette.white : palette.green.dark3};
+  `;
+
+  const upcomingStyles = css`
+    color: ${isDarkMode ? palette.gray.light1 : palette.gray.dark1};
+  `;
+
+  const upcomingMultipleStyles = css`
+    text-decoration-line: underline;
+    text-decoration-style: dotted;
+    text-underline-position: under;
+    color: ${isDarkMode ? palette.gray.light1 : palette.gray.dark1};
+  `;
+
+  const styles: Record<StepStates, string> = {
```

Comment:
🤔 I'd maybe write this outsize the component as a `Record<StepStates, Record<Mode, string>>` (I know we've discussed potential better ways to do doubly conditional styles)

Reply:
Yeah it feels related to the comment on the `InternalStep` - it feels kinda verbose to me and a little harder to compose styles.

```
const completedMultipleStyles = css`
  color: ${isDarkMode ? palette.green.base : palette.green.dark2};
`

const styles: Record<StepStates, string> = {
  [StepStates.CompletedMultiple]: completedMultipleStyles,
}
```
vs.

```
const darkModeCompletedMultipleStyles = css`
  color: ${palette.green.base};
`

const lightModeCompletedMultipleStyles = css`
  color: ${palette.green.dark2};
`

const styles: Record<StepStates, string> = {
  [Mode.Dark]: {
    [StepStates.CompletedMultiple]: darkModeCompletedMultipleStyles,
  },
  [Mode.Light]: {
    [StepStates.CompletedMultiple]: lightModeCompletedMultipleStyles,
  }
}
```
Not to mention - what if dark mode and light mode share some styles? Then in the second option, we'd need to have a separate `baseCompletedMultipleStyles` to compose in `cx`.